### ISSUE TITLE
LSP hover window lingers if you jump to definition

### ISSUE THREAD

**Issue by DarkDefender** [neovim/neovim#11395]
- `nvim --version`: NVIM v0.5.0-159-ge3b08a0fc-dirty
- `vim -u DEFAULTS` (version: ) behaves differently?
- Operating system/version: Gentoo Linux
- Terminal name/version: Alacritty 0.3.4
- `$TERM`: xterm-256color

### Steps to reproduce
- Use [nvim-lsp](https://github.com/neovim/nvim-lsp) with clangd and open up a C/CPP project.
- Activate the `lsp#text_document_hover()` on a function
- While the hover window is open, can any LSP function that jumps to a document like `lsp#text_document_definition()`

![output](https://user-images.githubusercontent.com/312503/68942006-ab1da100-07a7-11ea-83b6-ed4328343348.gif)

### Actual behaviour
The hover window stays in place and doesn't close until you go back to the original buffer

### Expected behaviour
The hover window should close when jumping around like this.

**Comment by e-kwsm** [comment]
Are the files—where the function is called and hover is opened, and where the function is defined—same? In my environment (`v0.5.0-171-gb1c4a8191` + pyls), the hover is closed when jumping to a function from the same file.
Actually, hover is not closed when `:e other-file` while opening hover.

**Comment by DarkDefender** [comment]
Yes, if you jump to the definition in the same file, it closes. This is only when jumping to other files.

**Comment by norcalli** [comment]
This should have been fixed with the most recent LSP changes (but do note that the vim interface has been changed in favor of using `v:lua.vim.lsp.buf.hover()` and friends).

**Comment by norcalli** [comment]
Btw, you can always use `<c-w>o` to hide floating windows, just in case you didn't know.

**Comment by DarkDefender** [comment]
Yes, this seems to be fixed now with the latest changes! Thanks! :)

**Comment by smhc** [comment]
Is this actually fixed? It seems to still be an issue with Neovim 0.5.1.

Using
```
  vim.lsp.handlers["textDocument/publishDiagnostics"] =
      vim.lsp.with(
      vim.lsp.diagnostic.on_publish_diagnostics,
      {
          underline = true,
          virtual_text = false,
          update_in_insert = true
      }
  )

  vim.cmd [[autocmd CursorHold * lua vim.lsp.diagnostic.show_line_diagnostics({focusable=false})]]
```

with clangd and the 'hover errors' are still there if you jump to definition while the float is displayed.

**Comment by smhc** [comment]
It looks like this fix got reverted somehow: https://github.com/neovim/neovim/pull/12477

**Comment by mjlbach** [comment]
I can't reproduce this.

1. Open src/nvim/main.c
2. Insert `foo` after input_init() on line 122 (before the semicolon)
3. put cursor over input_init
4. `vsp`
5. `:lua vim.diagnostic.open_float()`
6. `:lua vim.lsp.buf.definition()

**Comment by zeertzjq** [comment]
This has been fixed by #16557
